libparsec/crates/client/src/certif/device.rs

```diff
@@ -0,0 +1,175 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::sync::Arc;
+
+use libparsec_client_connection::{AuthenticatedCmds, ConnectionError};
+use libparsec_platform_storage::certificates::GetCertificateError;
+use libparsec_protocol::authenticated_cmds::v4::device_create;
+use libparsec_types::{
+    anyhow, thiserror, Bytes, CertificateSignerOwned, DateTime, DeviceCertificate, LocalDevice,
+    MaybeRedacted, SigningKeyAlgorithm, UserID,
+};
+
+use super::{
+    encrypt::CertifEncryptForUserError, greater_timestamp, CertificateBasedActionOutcome,
+    GreaterTimestampOffset, InvalidCertificateError,
+};
+
+pub async fn register_new_device(
+    cmds: Arc<AuthenticatedCmds>,
+    new_device: LocalDevice,
+    author: Arc<LocalDevice>,
+) -> Result<CertificateBasedActionOutcome, CertifDeviceError> {
+    // Loop is needed to deal with server requiring greater timestamp
+    let mut timestamp = author.now();
+
+    loop {
+        let outcome = internal_register_new_device(
+            cmds.clone(),
```

Comment:
I think you should be able to pass those parameters by reference fine instead of using clone here